mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -145,13 +135,20 @@ class LogisticRegressionModel (
        */
       var bestClass = 0
       var maxMargin = 0.0
-      var i = 0
-      while(i < margins.size) {
-        if (margins(i) > maxMargin) {
-          maxMargin = margins(i)
+      val withBias = if (dataMatrix.size + 1 == dataWithBiasSize) true else false
+      (0 until numClasses - 1).map { i =>
+        var margin = 0.0
+        dataMatrix.foreachActive { (index, value) =>
+          if (value != 0.0) margin += value * weightsArray((i * dataWithBiasSize) + index)
```

Comment:
@srowen  Thank you for your comments. I'm not sure that it will make a measurable difference, what I made for this PR is just want to clean code and reduce redundant. This optimization is inspired by #5137 which implements the prediction for Python API and it concerned mainly on code clean rather than performance issue.
I agree the real optimization may be not computing the weights array every time  prediction is made,  but just like you said it seems like small improvements. I think we should resolve it when we encounter some performance issues and file another ticket.